src/main/java/skylark/command/Command.java

```diff
@@ -0,0 +1,329 @@
+            stringBuilder.append(System.lineSeparator());
+            for (int i = 0; i < taskList.size(); i++) {
+                Task currentTask = taskList.get(i);
+                stringBuilder.append(i + 1).append(". ").append(currentTask.toString());
+                stringBuilder.append(System.lineSeparator());
+            }
+
+            return stringBuilder.toString();
+        }
+    }
+
+    private static class DoneCommand extends Command {
+        /**
+         * Returns a DoneCommand object.
+         *
+         * @param input String that is input by the user.
+         */
+        public DoneCommand(String input) {
+            super(input);
+        }
+
+        @Override
+        public String run(TaskList taskList) throws SkylarkException {
+            int index = Integer.parseInt(super.getInput().substring(5)) - 1;
+            String response;
+            if (taskList.doesIndexExist(index)) {
+                Task currentTask = taskList.get(index);
+                currentTask.markAsDone();
+                response = "Nice! I've marked this task as done:"
+                        + System.lineSeparator()
+                        + currentTask;
+                taskList.saveToFile();
+            } else {
+                throw new SkylarkException("Sorry, index does not exist!");
+            }
+
+            return response;
+        }
+    }
+
+    private static class UndoneCommand extends Command {
+        /**
+         * Returns a UndoneCommand object.
+         *
+         * @param input String that is input by the user.
+         */
+        public UndoneCommand(String input) {
+            super(input);
+        }
+
+        @Override
+        public String run(TaskList taskList) throws SkylarkException {
+            int index = Integer.parseInt(super.getInput().substring(7)) - 1;
+            String response;
+            if (taskList.doesIndexExist(index)) {
+                Task currentTask = taskList.get(index);
+                currentTask.markAsUndone();
+                response = "OK, I've marked this task as not done yet:"
+                        + System.lineSeparator()
+                        + currentTask;
+                taskList.saveToFile();
+            } else {
+                throw new SkylarkException("Sorry, index does not exist!");
+            }
+
+            return response;
+        }
+    }
+
+    private static class TodoCommand extends Command {
+        /**
+         * Returns a Todo object.
+         *
+         * @param input String that is input by the user.
+         */
+        public TodoCommand(String input) {
+            super(input);
+        }
+
+        @Override
+        public String run(TaskList taskList) throws SkylarkException {
+            if (super.getInput().equals(CommandList.COMMAND_TODO.toString())) {
+                throw new SkylarkException("The description of a todo cannot be empty.");
+            }
```

Comment:
Understand that you're checking for empty descriptions here (for deadline and event as well), but what if the input is "todo          " (however many spaces)?